src/pickup.cpp

```diff
@@ -169,6 +181,44 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
             veh_tool( "hotplate" );
             return DONE;
 
+        case USE_WASHMACHINE: {
+            bool detergent_is_enough = g->u.crafting_inventory().has_charges( "detergent", 5 );
+            bool filthy_items = false;
+            for( auto &i : veh->get_items( veh->part_with_feature( veh_root_part, "WASHING_MACHINE" ) ) ) {
+                static const std::string filthy( "FILTHY" );
+                if( i.has_flag( filthy ) ) {
+                    i.bday = calendar::turn.get_turn();
```

Comment:
Because items in the washing machine (`i`) are only accessible inside `for` cycle above, but it needed to be set below, in the `else if( !filthy_items )` check.